### PR TITLE
Correct implementation of setters/getters or input/output derivatives 

### PR DESCRIPTION
**Pull request by sharanry**
Tests are missing for these functions. I don't think the current set of test FMUs in `/models` can be used for testing.

TODO:

- [x] Correct the implementation of `fmi2GetRealOutputDerivatives`.
- [x] Correct the implementation of `fmi2SetRealInputDerivatives`.
- [x] Add tests.
- [x] Add docstrings.

### CODE REVIEW

**Comment by ThummeTo** [comment]
Dear @sharanry,
thank you for pointing on that! As I see, you are already on it.
Please let me know, if you need any assistance!
Thanks again for contributing!

Without testing, I think the Dymola/Windows-FMUs could support this, because the Dymola-Exporter is targeted on supporting the full FMI-standard.

Soon, a nice holiday!

File: test/getter_setter.jl [hunk 1]
```diff
@@ -119,6 +119,14 @@ fmiGetBoolean!(fmuStruct, booleanValueReferences, cacheBoolean)
 fmiGetString!(fmuStruct, stringValueReferences, cacheString)
 @test cacheString == rndString
 
+if envFMUSTRUCT == "FMUCOMPONENT"
+    dirs = fmi2GetRealOutputDerivatives(fmuStruct, realValueReferences, ones(2))
+    @test dirs isa AbstractArray{<:Real}
+    @test dirs |> length == 2
+
+    st = fmi2SetRealInputDerivatives(fmuStruct, realValueReferences, ones(2), zeros(2))
+    @test_broken st == FMI.fmi2OK
```
Comment (sharanry) [inline test/getter_setter.jl:128]: @ThummeTo I don't think even the dymola test fmus support input/output derivatives. `fmi2GetRealOutputDerivatives` returns zeros and `fmi2SetRealInputDerivatives` returns `fmi2Discarded`.

Comment (ThummeTo) [reply test/getter_setter.jl:128]: Dymola supports this, but there is no FMU that needs I/O-derivatives (no keywords in the model description). I can provide a simple FMU, that needs I/O-derivatives for automated testing. I opened the branch "i-o-derivatives", so we can merge there? (just change your PR from "main" to "i-o-derivatives").

Comment (sharanry) [reply test/getter_setter.jl:128]: Just did that.

**Pull request merged**